x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/EnterpriseSearchUsageTransportAction.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.application;
+
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.action.support.ActionFilters;
+import org.elasticsearch.client.internal.Client;
+import org.elasticsearch.cluster.ClusterState;
+import org.elasticsearch.cluster.metadata.IndexNameExpressionResolver;
+import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.common.inject.Inject;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.license.XPackLicenseState;
+import org.elasticsearch.logging.LogManager;
+import org.elasticsearch.logging.Logger;
+import org.elasticsearch.protocol.xpack.XPackUsageRequest;
+import org.elasticsearch.tasks.Task;
+import org.elasticsearch.threadpool.ThreadPool;
+import org.elasticsearch.transport.TransportService;
+import org.elasticsearch.xpack.application.search.action.ListSearchApplicationAction;
+import org.elasticsearch.xpack.application.utils.LicenseUtils;
+import org.elasticsearch.xpack.core.XPackSettings;
+import org.elasticsearch.xpack.core.action.XPackUsageFeatureAction;
+import org.elasticsearch.xpack.core.action.XPackUsageFeatureResponse;
+import org.elasticsearch.xpack.core.action.XPackUsageFeatureTransportAction;
+import org.elasticsearch.xpack.core.action.util.PageParams;
+import org.elasticsearch.xpack.core.application.EnterpriseSearchFeatureSetUsage;
+
+import java.util.Collections;
+import java.util.Map;
+
+public class EnterpriseSearchUsageTransportAction extends XPackUsageFeatureTransportAction {
+    private static final Logger logger = LogManager.getLogger(EnterpriseSearchUsageTransportAction.class);
+    private final XPackLicenseState licenseState;
+    private final Client client;
+
+    private final boolean enabled;
+
+    @Inject
+    public EnterpriseSearchUsageTransportAction(
+        TransportService transportService,
+        ClusterService clusterService,
+        ThreadPool threadPool,
+        ActionFilters actionFilters,
+        IndexNameExpressionResolver indexNameExpressionResolver,
+        Settings settings,
+        XPackLicenseState licenseState,
+        Client client
+    ) {
+        super(
+            XPackUsageFeatureAction.ENTERPRISE_SEARCH.name(),
+            transportService,
+            clusterService,
+            threadPool,
+            actionFilters,
+            indexNameExpressionResolver
+        );
+        this.licenseState = licenseState;
+        this.client = client;
```

Comment:
Calling the action (list search application) directly requires to have the `manage_search_application` privilege. 
We can authorise this specific action to always have access to the search application internals by using the [`ent search origin`](https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ClientHelper.java#L193):
```
this.client = new OriginSettingClient(client, ENT_SEARCH_ORIGIN);
```
The rest tests are working in this PR because the test user has the [`manage_search_application` privilege](https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/ent-search/qa/rest/roles.yml#L3). Using the ent-search origin will ensure that this privilege is no longer required.